src/packages/@ncigdc/components/Charts/BarChart.js

```diff
@@ -145,56 +149,84 @@ const BarChart = ({
-    .call(d3.axisBottom(x));
-
-  hiddenXG.selectAll('text')
-    .style('text-anchor', 'start')
-    .style('fontSize', xAxisStyle.fontSize)
-    .style('fontWeight', xAxisStyle.fontWeight)
-    .attr('fill', xAxisStyle.textFill)
-    .attr('dx', '.8em')
-    .attr('dy', '.5em')
-    .text(d => d)
-    .attr('transform', 'rotate(45)');
-
-  hiddenXG.selectAll('path').style('stroke', xAxisStyle.stroke);
-    
-  hiddenXG.selectAll('line').style('stroke', xAxisStyle.stroke);
-
-  // display x axis
-  const xG = svg
-    .append('g')
-    .attr('class', 'displayOnly')
-    .attr('transform', `translate(0, ${height})`)
-    .call(d3.axisBottom(x));
+  if (showXAxisLabels) {
+    // hidden x axis for full length labels in svg download
+    const hiddenXG = svg
+      .append('g')
+      .attr('class', 'svgDownload')
+      .attr('transform', `translate(0, ${height})`)
+      .call(d3.axisBottom(x));
 
-  xG.selectAll('text')
-    .style('text-anchor', 'start')
-    .style('fontSize', xAxisStyle.fontSize)
-    .style('fontWeight', xAxisStyle.fontWeight)
-    .attr('fill', xAxisStyle.textFill)
-    .attr('dx', '.8em')
-    .attr('dy', '.5em')
-    .text(d => (d.length > xAxisLabelLength ? `${d.substring(0, xAxisLabelLength - 3)}...` : d))
-    .attr('transform', 'rotate(45)');
+    hiddenXG.selectAll('text')
+      .style('text-anchor', 'start')
+      .style('fontSize', xAxisStyle.fontSize)
+      .style('fontWeight', xAxisStyle.fontWeight)
+      .attr('fill', xAxisStyle.textFill)
+      .attr('dx', '.8em')
+      .attr('dy', '.5em')
+      .text(d => d)
+      .attr('transform', 'rotate(45)');
 
-  xG.selectAll('path').style('stroke', xAxisStyle.stroke);
+    hiddenXG.selectAll('path').style('stroke', xAxisStyle.stroke);
 
-  xG.selectAll('line').style('stroke', xAxisStyle.stroke);
+    hiddenXG.selectAll('line').style('stroke', xAxisStyle.stroke);
 
-  xG.selectAll('.tick')
-    .data(data)
-    .on('mouseenter', d => {
-      setTooltip(d.tooltip);
-    })
-    .on('mouseleave', () => {
-      setTooltip();
-    });
+    // display x axis
+    const xG = svg
+      .append('g')
+      .attr('class', 'displayOnly')
+      .attr('transform', `translate(0, ${height})`)
+      .call(d3.axisBottom(x));
+
+    xG.selectAll('text')
+      .style('text-anchor', 'start')
+      .style('fontSize', xAxisStyle.fontSize)
+      .style('fontWeight', xAxisStyle.fontWeight)
+      .attr('fill', xAxisStyle.textFill)
+      .attr('dx', '.8em')
+      .attr('dy', '.5em')
+      .text(d => (d.length > xAxisLabelLength ? `${d.substring(0, xAxisLabelLength - 3)}...` : d))
+      .attr('transform', 'rotate(45)');
+
+    xG.selectAll('path').style('stroke', xAxisStyle.stroke);
+
+    xG.selectAll('line').style('stroke', xAxisStyle.stroke);
+
+    xG.selectAll('.tick')
+      .data(data)
+      .on('mouseenter', d => {
+        setTooltip(d.tooltip);
+      })
+      .on('mouseleave', () => {
+        setTooltip();
+      });
+  } else {
```

Comment:
After trying to find the commit in which said `if` is removed, I've noticed the if was in fact added directly to develop when integrating summary page, which is why I've added it here.
https://github.com/NCI-GDC/portal-ui/commit/d28584433e1a56e9b74a8867460d5f756713b82c#diff-d9be36a0023b28d43a6dd20f137e77c4R98

May this be some disparity from the branching, in which a "better" version of the code was added to the release, rather than what would've been an incremental improvement, had we been able to keep the branches synced?

Ultimately, if I understand correctly, you suggest removing that whole `if`, as the other `else` handles the same purpose/requirement?

Reply:
the else was added here https://github.com/NCI-GDC/portal-ui/commit/8a425d51fdf8da5378ade146d4d5959eb8e8db2a  but i don't know what commit the other `if (xAxis.title)` was supposed to be removed in, i don't see it in that diff

Reply:
That's correct. it wasn't removed (from develop, where it's coming from, and thus not showing in the diff); therefor, it is still present.
Should we just remove it? I'm not acquainted enough with these lines' purpose.

Reply:
yes i think it should be removed. Looks like it was copied/pasted into the new `else`

Reply:
hm... on second thought, we should also make that `else` an `else if` and check for that `title` attribute being present. Else we may see an "undefined" issue.

Reply:
ehrm no. that https://github.com/NCI-GDC/portal-ui/commit/8a425d51fdf8da5378ade146d4d5959eb8e8db2a also adds a check in line 35.
Ok, removing the `if`. Nice catch, @anncatton!